src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -675,4 +676,22 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 			this._modelToDisposeMap.set(editor.id, listeners);
 		});
 	}
+
+	$registerNavigationProvider(providerId: string, handle: number) {
+		this._notebookService.registerNavigationProvider({
+			providerId: providerId,
+			onNext: async (uri) => {
```

Comment:
We need 2 additional things here - are you adding in a follow up review?
1. HasNavigation so that we can decide whether to show the buttons at all
2. status (enabled / disabled) for the previous and next buttons so we can disable if there is no navigation.

As mentioned, you can do in a smaller follow up if needed.

Reply:
Clarified offline - will be covered in separate PRs.